src/main/java/org/primeframework/mvc/message/l10n/ResourceBundleMessageProvider.java

```diff
@@ -120,21 +121,22 @@ protected Queue<String> determineBundles(String bundle) {
 
   /**
    * Finds the message in a resource bundle using the search method described in the class comment.
+   * If the action was annotated with a {@link org.primeframework.mvc.action.annotation.MessageResources} annotation
+   * and the message was not found with the request's action, that action will be searched as well.
    *
    * @param actionInvocation The action invocation.
    * @param key              The key of the message.
    * @return The message or null if it doesn't exist.
    */
   protected String findMessage(ActionInvocation actionInvocation, String key) {
-    String uri = actionInvocation.actionURI;
-    Queue<String> names = determineBundles(uri);
-    for (String name : names) {
-      try {
-        ResourceBundle rb = ResourceBundle.getBundle(name, localeProvider.get(), control);
-        return rb.getString(key);
-      } catch (MissingResourceException ignore) {
-        // Ignore and check the next bundle
-      }
+    String message = findMessage(actionInvocation.actionURI, key);
+    if (message != null) {
+      return message;
+    }
+
+    ActionConfiguration config = actionInvocation.configuration;
+    if (config != null && config.fallbackMessageResourcesPath != null) {
```

Comment:
I really dislike checking this for null. I'd rather bake in the assumption that we always have an action config. But it was much simpler test wise to do this.

Reply:
I think you can pull this off. See my comments in the annotation, but if it is an array, then this config would be an empty collection or not empty, never null.

Reply:
the problem aint the fallback path, it's the config itself

Reply:
Partially Fixed.